Source/utils/paths.cpp

```diff
@@ -75,7 +76,10 @@ const std::string &PrefPath()
 {
 	if (!prefPath)
 		prefPath = FromSDL(SDL_GetPrefPath("diasurgical", "devilution"));
-	return *prefPath;
+	if (FileExists("diablo.ini")) {
+		prefPath = "./";
+	}

```

Comment:
Looks like this path doesn't care if the folder is writable or not. Maybe move the check earlier so that it can influence both paths and only has to run once. But also make sure that if the path is given specifically by the user then we do not run this check.

Reply:
I rewrote the save one so it checks for write access and if the user set the path.

But I couldn't figure out where exactly I would run it earlier, because they both return to static and const. Which those can't be written twice can they?

Reply:
Got it. Done.